spec/controllers/api/v3/twilio_sms_delivery_controller_spec.rb

```diff
@@ -97,49 +97,90 @@
           appointment_reminder = create(:appointment_reminder)
-          communication = create(:communication, :missed_visit_whatsapp_reminder, appointment_reminder: appointment_reminder)
+          create(:communication, :missed_visit_whatsapp_reminder, appointment_reminder: appointment_reminder)
+          communication = create(:communication, :missed_visit_sms_reminder, appointment_reminder: appointment_reminder)
           create(:twilio_sms_delivery_detail, session_id: session_id, result: "queued", communication: communication)
 
-          allow(Communication).to receive(:next_messaging_time).and_return(fallback_time)
-
           params = base_callback_params.merge(
             "MessageSid" => session_id,
             "MessageStatus" => "failed"
           )
+          set_twilio_signature_header(callback_url, params)
 
-          expect(AppointmentNotification::Worker).to receive(:perform_at).with(
-            fallback_time,
-            communication.appointment_reminder_id,
-            "missed_visit_sms_reminder"
-          )
+          expect(AppointmentNotification::Worker).not_to receive(:perform_at)
+          expect(Statsd.instance).to receive(:increment).with("twilio_callback.missed_visit_sms_reminder.failed")
 
-          set_twilio_signature_header(callback_url, params)
           post :create, params: params
         end
 
-        it "does not schedule a fallback SMS if an SMS failed" do
+        it "logs failure and schedules a new attempt if the first attempt failed and there is a next communication type" do
           session_id = SecureRandom.uuid
           fallback_time = 5.minutes.from_now
-          communication = create(:communication, :missed_visit_sms_reminder)
+          appointment_reminder = create(:appointment_reminder)
+          communication = create(:communication, :missed_visit_whatsapp_reminder, appointment_reminder: appointment_reminder)
           create(:twilio_sms_delivery_detail, session_id: session_id, result: "queued", communication: communication)
 
-          allow(Communication).to receive(:next_messaging_time).and_return(fallback_time)
-
           params = base_callback_params.merge(
             "MessageSid" => session_id,
             "MessageStatus" => "failed"
           )
+          set_twilio_signature_header(callback_url, params)
 
-          expect(AppointmentNotification::Worker).not_to receive(:perform_at)
+          allow(Communication).to receive(:next_messaging_time).and_return(fallback_time)
+          expect(AppointmentNotification::Worker).to receive(:perform_at).with(
+            fallback_time,
+            communication.appointment_reminder_id
+          )
+          expect(Statsd.instance).to receive(:increment).with("twilio_callback.missed_visit_whatsapp_reminder.failed")
 
-          set_twilio_signature_header(callback_url, params)
           post :create, params: params
         end
+
+        # it "schedules a fallback SMS if a whatsapp message failed" do
```

Comment:
Why is all this commented out?

Reply:
This is an oopsie daisy. It was the lat thing I changed and I obviously failed to delete the old code.